tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,290 @@
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Waiting for pod to be ready")
+					waitForPodReadiness(podName, namespace, 1*time.Minute)
+				})
+
+				AfterEach(func() {
+					err := clientset.CoreV1().Pods(namespace).Delete(podName, nil)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("SHOULD successfully get a second interface, of macvtap type, with configured MAC address", func() {
+					pod, err := clientset.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					// assert resources have been allocated
+					Expect(pod.Annotations).To(HaveKey(networkResource))
+					Expect(pod.Spec.Containers).To(HaveLen(1))
+
+					theContainer := pod.Spec.Containers[0]
+					Expect(theContainer.Resources.Limits).To(Equal(buildMacvtapResourceRequest(lowerDevice, 1)))
+
+					// assert MAC address is found on the second interface
+					podNetworks := pod.Annotations[networkStatus]
+					networks, err := parseNetwork(podNetworks)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(networks).To(HaveLen(2))
+
+					// macvtap iface is created as a secondary iface
+					macvtapNetwork := networks[1]
+					Expect(macvtapNetwork.Name).To(Equal(macvtapIfaceName))
+					Expect(macvtapNetwork.Interface).To(Equal("net1"))
+					Expect(macvtapNetwork.Mac).NotTo(BeEmpty())
+				})
+			})
+
+			PContext("WHEN a VMI requests aforementioned macvtap resource", func() {
+				It("THEN the macvtap interface can be used to reach the internet", func() {
+
+				})
+
+				Context("WHEN another VMI is created, having an additional macvtap interface", func() {
+					It("THEN it can reach another VMI via the macvtap interface", func() {
+
+					})
+
+					It("THEN it cannot reach the host where its pod was scheduled", func() {
+
+					})
+
+					It("THEN it can reach the other hosts in the cluster", func() {
+
+					})
+				})
+			})
+		})
+	})
+})
+
+func deleteNetworkAttachmentDefinition(macvtapIfaceName string, namespace string) rest.Result {
+	return clientset.RESTClient().
+		Delete().
+		RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+		Do()
+}
+
+func provisionNetworkAttachmentDefinition(macvtapIfaceName string, lowerDeviceName string, namespace string) rest.Result {
+	return clientset.RESTClient().
+		Post().
+		RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+		Body([]byte(fmt.Sprintf(nad, macvtapIfaceName, namespace, buildMacvtapResourceName(lowerDeviceName), macvtapIfaceName))).
+		Do()
+}
+
+func filterPods(pods []v1.Pod, filterFunction func(v1.Pod) bool) []v1.Pod {
+	filteredPods := make([]v1.Pod, 0)
+	for _, pod := range pods {
+		if filterFunction(pod) {
+			filteredPods = append(filteredPods, pod)
+		}
+	}
+	return filteredPods
+}
+
+func buildMacvtapResourceName(macvtapIfaceName string) string {
+	return fmt.Sprintf("%s/%s", macvtapResource, macvtapIfaceName)
+}
+
+func waitForResourceAvailability(timeout time.Duration, resourceName string) {
```

Comment:
I was also confused with this naming for a bit, in the sense of kubernetes resources vs compute resources; and I thought that I would rename this to waitForNodeResourceAvailability.

Reply:
I think we're consistent enough in this PR / repo about what a resource is. 

Never the less, I'm all for clarity; no problem in renaming it to what you suggest if it improves readability / clarity.